libjcat/jcat-self-test.c

```diff
@@ -576,6 +581,36 @@ jcat_pkcs7_engine_self_signed_func(void)
 	str = jcat_result_to_string(result);
 	g_print("%s", str);
 	g_assert_true(g_pattern_match_simple(str_perfect, str));
+
+	/* verify a payload with the wrong signature */
+	payload_other = g_bytes_new_static(payload_other_str, sizeof(payload_other_str));
+	result_fail = jcat_engine_self_verify(engine,
+					      payload_other,
+					      jcat_blob_get_data(signature),
+					      JCAT_VERIFY_FLAG_NONE,
+					      &error);
+	g_assert_error(error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA);
+	g_assert_null(result_fail);
+	g_clear_error(&error);
+
+	/* change keyring path */
+	tmp_dir = g_dir_make_tmp(NULL, &error);
+	g_assert_no_error(error);
+	jcat_context_set_keyring_path(context, tmp_dir);
```

Comment:
I think we should do this for the other test too; i.e. we probably use a unique directory for every run.

Reply:
I was trying to figure out how much time is added to each test run by not sharing a keyring path.
I ran it too many times and meson started timing out the tests after 30 seconds: `ls -d /tmp/.*/**/{gnupg,pki}/ | wc -l` returned 674.

It's probably a good idea to clean up the directories at the end of each successful test.

Also would there be a downside to creating a shared keyring path for a batch of tests in `main`?

On my machine the last commit brings minimum time of a test run up from 0.09s to 0.15s but I think there's much more variability.

Reply:
> It's probably a good idea to clean up the directories at the end of each successful test.

Right; good catch -- this is what I do in fwupd: https://github.com/fwupd/fwupd/blob/main/libfwupdplugin/fu-path.c#L34

> Also would there be a downside to creating a shared keyring path for a batch of tests

You could do, but I think there's value in being able to run them one by one too. If you're doing it using a test harness that's fine, less so if you're relying on order.

> test run up from 0.09s to 0.15s

Literally fine.

Reply:
TIL!